### ISSUE TITLE
Random seed in tf.keras.initializers.RandomNormal does not work in Tensorflow 2.7.0

### ISSUE THREAD

**Issue by lubin-liu** [keras-team/keras#15755]
Please go to TF Forum for help and support:

https://discuss.tensorflow.org/tag/keras

If you open a GitHub issue, here is our policy:

It must be a bug, a feature request, or a significant problem with the documentation (for small docs fixes please send a PR instead).
The form below must be filled out.

**Here's why we have that policy:**.

Keras developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Google Colab
- TensorFlow installed from (source or binary): 
- TensorFlow version (use command below): 2.7.0
- Python version: 3.7.12
- Bazel version (if compiling from source): N/A
- GPU model and memory: N/A
- Exact command to reproduce: N/A

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with:
python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"

**Describe the problem**.

Describe the problem clearly here. Be sure to convey here why it's a bug in Keras or why the requested feature is needed.

**Describe the current behavior**.

Two instantiations of tf.keras.initializers.RandomNormal(..) objects with the same seed does not produce the same sequence.

**Describe the expected behavior**.

From the documentation [here](https://www.tensorflow.org/api_docs/python/tf/keras/initializers/RandomNormal):

```
 tf.keras.initializers.RandomNormal(
     mean=0.0, stddev=0.05, seed=None
 )
```

Regarding the seed parameter, the documentation states: 

> ... Note that seeded initializer will not produce same random values across multiple calls, but multiple initializers will produce same sequence when constructed with same seed value. 

I would expect that if I pass the same seed as a parameter into the two instantiations of initializers.RandomNormal, temp and temp2, both should give the same weights on the first call, which is what I'm observing in Tensorflow 2.3.0, 2.5.0, and 2.6.0 ([example](https://colab.research.google.com/drive/10KrXpJtxiaAmnI2vKa61hDNheqjiJyu9?usp=sharing)), but not 2.7.0.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no): No
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**.

Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

Using tensorflow version 2.7.0:
https://colab.research.google.com/drive/1paNe8kBRzwkrZHBi7QF0yhEX8pXO0f1w?usp=sharing

However, using tensorflow 2.6.0, this issue does not occur (note one has to restart the notebook after the pip install in the first cell before running the second cell):
https://colab.research.google.com/drive/10KrXpJtxiaAmnI2vKa61hDNheqjiJyu9?usp=sharing

**Source code / logs**.

Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by foxik** [comment]
Yes, I have also observed this behaviour.

The problem is that the new codepath using the `tf.random.Generator` is not enabled:
https://github.com/keras-team/keras/blob/5d2993b78603bda911f0a4912d08b1e5f36eb633/keras/backend.py#L1752-L1757

so the legacy methods like `tf.random.normal` are used instead https://github.com/keras-team/keras/blob/5d2993b78603bda911f0a4912d08b1e5f36eb633/keras/backend.py#L1864-L1872
and these are stateful and do not return the same values for the same seeds.

Note that as a hack, you can "switch" to the new behaviour by using
```python
import keras
keras.backend.enabled_generator_for_rng()
```

**Comment by foxik** [comment]
Adding @qlzh727 who authored the commits.

Note that the current behaviour is contradicting the documentation of the initializers
https://github.com/keras-team/keras/blob/5d2993b78603bda911f0a4912d08b1e5f36eb633/keras/initializers/initializers_v2.py#L257-L261

stating that
> "but multiple initializers will produce same sequence when constructed with same seed value"

which is quite unfortunate.

**Comment by qlzh727** [comment]
See https://github.com/keras-team/keras/issues/15586 for more details. The short answer is that you either need to force the generator code path, which we will turn on by default soon, or you need to call tf.random.set_seed() in addition to make the generator to produce same sequence.

**Comment by qlzh727** [comment]
Close this bug since this is a duplicate with https://github.com/keras-team/keras/issues/15586
